wavespectra/input/spotter.py

```diff
@@ -6,16 +6,206 @@
+    ds['Latitude (deg)'].attrs['standard_name'] = 'latitude'
+    ds['Latitude (deg)'].attrs['long_name'] = 'Latitude'
+
+    ds['Longitude (deg)'].attrs['units'] = 'degree_east'
+    ds['Longitude (deg)'].attrs['standard_name'] = 'longitude'
+    ds['Longitude (deg)'].attrs['long_name'] = 'Longitude'
+
+    ds['Wind Speed (m/s)'].attrs['units'] = 'm/s'
+    ds['Wind Speed (m/s)'].attrs['standard_name'] = 'wind_speed'
+    ds['Wind Speed (m/s)'].attrs['long_name'] = 'Wind speed'
+
+    ds['Wind Direction (deg)'].attrs['units'] = 'degree'
+    ds['Wind Direction (deg)'].attrs['standard_name'] = 'wind_from_direction'
+    ds['Wind Direction (deg)'].attrs['long_name'] = 'Wind direction'
+
+    ds['Surface Temperature (°C)'] = 274.15*ds['Surface Temperature (°C)']
+    ds['Surface Temperature (°C)'].attrs['units'] = 'K'
+    ds['Surface Temperature (°C)'].attrs['standard_name'] = 'sea_surface_temperature'
+    ds['Surface Temperature (°C)'].attrs['long_name'] = 'Surface temperature'
+
+    ds['Frequency'].attrs['units'] = 'Hz'
+    ds['Frequency'].attrs['standard_name'] = 'wave_frequency'
+    ds['Frequency'].attrs['long_name'] = 'Frequency'
+
+    ds['Variance density'].attrs['units'] = 'm^2/Hz'
+    ds['Variance density'].attrs['standard_name'] = 'sea_surface_wave_variance_spectral_density'
+    ds['Variance density'].attrs['long_name'] = 'Spectral density'
+
+    ds['Directional spread'].attrs['units'] = 'degree'
+    ds['Directional spread'].attrs['standard_name'] = 'sea_surface_wave_directional_spread'
+    ds['Directional spread'].attrs['long_name'] = 'Directional spreading'
+    
+    ds.attrs['Conventions'] = 'CF-1.8'
+    ds.attrs['source'] = f"Sofar Spotter buoy, {filename}"
+    ds.attrs['history'] = f"generated {datetime.datetime.now().strftime('%Y-%m-%d @ %H:%M:%S')} by {getpass.getuser()}"
+    ds.attrs['references'] = 'https://content.sofarocean.com/hubfs/Technical_Reference_Manual.pdf'
+
+
+    ds = ds.rename({'Epoch Time':'time',
+                    'Frequency':'freq',
+                    'Battery Voltage (V)':'batter_voltage',
+                    'Variance density':'etfh',
+                    'Direction':'dir',
+                'Power (W)':'battery_power',
+                'Humidity (%rel)':'humidity',
+                'Significant Wave Height (m)':'Hm0',
+                'Peak Period (s)':'Tp',
+                'Mean Period (s)':'Tm',
+                'Peak Direction (deg)':'peak_dir',
+                'Peak Directional Spread (deg)':'peak_spread',
+                'Mean Direction (deg)':'mean_dir',
+                'Mean Directional Spread (deg)':'mean_spread',
+                'Directional spread':'spread',
+                'Latitude (deg)':'lat',
+                'Longitude (deg)':'lon',
+                'Wind Speed (m/s)':'wspd',
+                'Wind Direction (deg)':'wdir',
+                'Surface Temperature (°C)':'temperature'})
+    
+    ds = ds.sortby('time')
+    
+    return ds
```

Comment:
It looks like maybe I should separate data variables into `.dset` (which includes only the variables listed here: https://wavespectra.readthedocs.io/en/latest/conventions.html#data-variables) and `.spot` which contains anything else recorded in the `CSV` (e.g., `Battery Voltage`, `Significant Wave Height`, `Peak Period`, etc.)

Reply:
No, please see my comment above. You don't really touch the `.spec` accessor, that is created by the SpecArray / SpecDataset main classes themselves. All you have to do is return an xarray dataset with the correct conventions regarding coordinate naming and units.